cloud/linode/route_controller.go

```diff
@@ -0,0 +1,252 @@
+	if err != nil {
+		return err
+	}
+	for _, r := range resp {
+		vpcNodes[r.LinodeID] = append(vpcNodes[r.LinodeID], r)
+	}
+
+	rc.routes = vpcNodes
+	rc.lastUpdate = time.Now()
+	return nil
+}
+
+type routes struct {
+	vpcid      int
+	client     client.Client
+	instances  *instances
+	routeCache *routeCache
+}
+
+func newRoutes(client client.Client) (cloudprovider.Routes, error) {
+	timeout := 60
+	if raw, ok := os.LookupEnv("LINODE_ROUTES_CACHE_TTL"); ok {
+		if t, _ := strconv.Atoi(raw); t > 0 {
+			timeout = t
+		}
+	}
+	klog.V(3).Infof("TTL for routeCache set to %d", timeout)
+
+	vpcid := vpcInfo.getID()
+	if Options.EnableRouteController && vpcid == 0 {
+		return nil, fmt.Errorf("cannot enable route controller as vpc [%s] not found", Options.VPCName)
+	}
+
+	return &routes{
+		vpcid:     vpcid,
+		client:    client,
+		instances: newInstances(client),
+		routeCache: &routeCache{
+			routes: make(map[int][]linodego.VPCIP, 0),
+			ttl:    time.Duration(timeout) * time.Second,
+		},
+	}, nil
+}
+
+// instanceRoutesByID returns routes for given instance id
+func (r *routes) instanceRoutesByID(id int) ([]linodego.VPCIP, error) {
+	r.routeCache.RLock()
+	defer r.routeCache.RUnlock()
+	instanceRoutes, ok := r.routeCache.routes[id]
+	if !ok {
+		return nil, fmt.Errorf("no routes found for instance %d", id)
+	}
+	return instanceRoutes, nil
+}
+
+// getInstanceRoutes returns routes for given instance id
+// It refreshes routeCache if it has expired
+func (r *routes) getInstanceRoutes(ctx context.Context, id int) ([]linodego.VPCIP, error) {
+	if err := r.routeCache.refreshRoutes(ctx, r.client); err != nil {
+		return nil, err
+	}
+
+	return r.instanceRoutesByID(id)
+}
+
+// getInstanceFromName returns linode instance with given name if it exists
+func (r *routes) getInstanceFromName(ctx context.Context, name string) (*linodego.Instance, error) {
+	// create node object
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+
+	// fetch instance with specified node name
+	instance, err := r.instances.lookupLinode(ctx, node)
+	if err != nil {
+		klog.Errorf("failed getting linode %s", name)
+		return nil, err
+	}
+	return instance, nil
+}
+
+// CreateRoute adds route's subnet to ip_ranges of target node's VPC interface
+func (r *routes) CreateRoute(ctx context.Context, clusterName string, nameHint string, route *cloudprovider.Route) error {
+	instance, err := r.getInstanceFromName(ctx, string(route.TargetNode))
+	if err != nil {
+		return err
+	}
+
+	// fetch instance routes
+	instanceRoutes, err := r.getInstanceRoutes(ctx, instance.ID)
+	if err != nil {
+		return err
+	}
+
+	// check already configured routes
+	intfRoutes := []string{}
+	intfVPCIP := linodego.VPCIP{}
+	for _, ir := range instanceRoutes {
+		if ir.Address != nil && ir.VPCID == r.vpcid {
+			intfVPCIP = ir
+			continue
+		}
+
+		if ir.Address != nil || ir.VPCID != r.vpcid {
+			continue
+		}
```

Comment:
```suggestion
		if ir.VPCID != r.vpcid {
			continue
		}

		if ir.Address != nil {
			intfVPCIP = ir
			continue
		}
```

I think that having the logic laid out like this makes it a little easier to follow.

Reply:
Thanks, updated